src/contrib/mlir/lowerer.cpp

```diff
@@ -1029,6 +1007,27 @@ namespace
 
         rewriter.replaceOp(op, result);
     }
+
+    ValueHandle createZeroConstant(mlir::Type type)
+    {
+        if (auto floatTy = type.dyn_cast<FloatType>())
+        {
+            if (floatTy.isF32())
+                return intrinsics::constant_float(llvm::APFloat(0.0f), floatTy);
+            else if (floatTy.isF64())
+                return intrinsics::constant_float(llvm::APFloat(0.0), floatTy);
```

Comment:
What was the issue? We were creating a single precision constant for a double precision? Is APFloat constructor overloaded for float and double? Interesting...

Reply:
Yes